nntrainer/src/tensor.cpp

```diff
@@ -829,13 +841,29 @@ void Tensor::read(std::ifstream &file) {
  * @brief Calculate average value according to the axis.
  */
 Tensor Tensor::average(int axis) const {
+  if (axis < 0 || axis >= MAXDIM)
```

Comment:
better to use unsigned int? or do we support -1 like python?

Reply:
If we support a negative axis, I think it'd better be done in a different patch. I changed to unsigned int for now. Thanks.